### ISSUE TITLE
[help] Add multigpu evalulation script for seq2seq

### ISSUE THREAD

**Issue by sshleifer** [huggingface/transformers#6785]
### Goal
At the end the command,
```
cd examples/seq2seq
export DATA_DIR=test_data  # use wmt_en_ro for more data, download instructions in README.md
export save_dir=translations
python multigpu_eval.py Helsinki-NLP/opus-mt-en-ro $DATA_DIR/test.source $save_dir/test_translations.txt --reference_path $DATA_DIR/test.target --score_path $save_dir/test_bleu.json --fp16 --task translation --bs 16
```
On wmt_en_ro, thissShould get ~27.7 (same result as run_eval.py on 1 GPU) and should take less time.
On test_data I don't know the 1 GPU result.

side note: The motivation here is actually summarization, where the validation sets are larger and on 1 GPU for pegasus running eval takes an hour.

### Implementation

This looks similar, https://github.com/facebookresearch/ParlAI/blob/00efcbebb49524918692638ab580cadeebe70cf8/parlai/scripts/multiprocessing_eval.py#L34 but I'm open to many strategies. Lots of freedom. You will probably need to use one of the datasets in utils + an unshuffled dataloader of some sort. Try to keep the command line as similar to run_eval.py as possible, with the exception of a --gpus flag. Alternatively, use all available gpus and tell the user to use CUDA_VISIBLE_DEVICES=0,1 to restrict.

### Process

1. You can claim the bug here, but don't wait for me to reply to get started. This is hard enough that it is very unlikely that two people will be working on this at once. If I start working on this (unlikely) I will post here and remove the help-wanted label.
2. When you send a PR, tag me and link this issue. 
3. It will be difficult to test your solution in CI, but try to make the code work with 1 GPU so we can test it that way in CI. You can make a new test_.py file or add to test_seq2seq_examples.py
4. Report BLEU scores in PR description.

Happy hacking!
